experiments/react-native-uifabric/src/components/PersonaCoin/PersonaCoin.tsx

```diff
@@ -0,0 +1,169 @@
+import {
+  IPersonaCoinProps,
+  IPersonaCoinType,
+  personaCoinName,
+  IPersonaCoinSlotProps,
+  IPersonaCoinRenderData,
+  PersonaSize,
+  IPersonaCoinState
+} from './PersonaCoin.types';
+import { compose, IUseComposeStyling } from '@uifabricshared/foundation-compose';
+import { filterViewProps, filterImageProps } from '../../utilities/RenderHelpers';
+import { settings } from './PersonaCoin.settings';
+import { ISlots, withSlots, IRenderData } from '@uifabricshared/foundation-composable';
+import { mergeSettings } from '@uifabricshared/foundation-settings';
+import { getSizeConfig, convertCoinColor, getPresenceIconSource } from './PersonaCoin.helpers';
+import { buildPersonaCoinRootStyles, buildPersonaCoinContentStyles } from './PersonaCoin.tokens';
+import { foregroundColorTokens, backgroundColorTokens } from '../../tokens';
+
+function usePrepareForProps(
+  props: IPersonaCoinProps,
+  useStyling: IUseComposeStyling<IPersonaCoinType>
+): IRenderData<IPersonaCoinSlotProps, IPersonaCoinState> {
+
+  const { imageUrl, imageDescription, size, initials, coinColor, presence, ...rest } = props;
+
+  const useSizeFromProps = size !== undefined;
+  const normalizedSize = size === undefined ? PersonaSize.size40 : size;
+  const { physicalCoinSize, initialsFontSize, iconSize } = getSizeConfig(normalizedSize);
+
+  const sizeStyle: ImageStyle = {};
+  if (useSizeFromProps) { 
+    sizeStyle.width = physicalCoinSize;
+    sizeStyle.height = physicalCoinSize;
+  }
+
+  let personaPhotoSource: ImageURISource | undefined;
+  let photoStyle: ImageStyle | undefined;
+  let initialsStyle: TextStyle | undefined;
+  let initialsBackgroundStyle: ViewStyle | undefined;
+
+  if (imageUrl) {
+    personaPhotoSource = {
+      uri: imageUrl
+    };
+
+    photoStyle = {...sizeStyle};
+    if (useSizeFromProps) {
+      photoStyle.borderRadius = physicalCoinSize / 2;
+    }
+  } else {
+    initialsStyle = {};
+    initialsBackgroundStyle = { flexGrow: 1, alignSelf: 'stretch', justifyContent: 'center', alignItems: 'center' };
+
+    if (useSizeFromProps) {
+      initialsBackgroundStyle.borderRadius = physicalCoinSize / 2;
+      initialsStyle.fontSize = initialsFontSize;
+    }
+
+    if (coinColor !== undefined) {
+      initialsBackgroundStyle.backgroundColor = convertCoinColor(coinColor);
+    }
+  }
+
+  let iconSource: ImageURISource | undefined = undefined;
+  let iconStyle: ImageStyle | undefined = undefined;
+  if (presence) {
+    iconSource = getPresenceIconSource(presence);
+    iconStyle = {
+      position: 'absolute',
+    };
+
+    if (useSizeFromProps && iconSize > 0 ) {
+      iconStyle.width = iconSize;
+      iconStyle.height = iconSize;
+    }
+  }
+
+  return {
+    slotProps: mergeSettings<IPersonaCoinType['slotProps']>(useStyling(props), {
+      root: { ...rest, style: sizeStyle },
+      initials: {
+        children: initials,
+        style: initialsStyle

```

Comment:
This is a tricky one, you are passing a style object created inside this function into the settings, this will then get merged with the cached styles but will create a new object on each render pass.  As a result this control will re-render on each pass, even if it doesn't need to.  Granted the entry props for the persona block a lot of that but it still adds expense.

It might be better to use style functions in the styles section to produce the styles based on tokens.  This way the results will be cached based on token entry values.  It also means they will be shared across persona instances.  Come talk to me and I can explain how that might work.